vault-rcp/src/main/java/org/apache/jackrabbit/vault/rcp/impl/FileNodeBackedOutputStream.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.jackrabbit.vault.rcp.impl;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+
+import javax.jcr.Node;
+import javax.jcr.RepositoryException;
+import javax.jcr.Session;
+
+import org.apache.jackrabbit.JcrConstants;
+import org.apache.jackrabbit.commons.JcrUtils;
+import org.apache.jackrabbit.vault.util.Text;
+import org.apache.sling.jcr.api.SlingRepository;
+
+public class FileNodeBackedOutputStream extends OutputStream {
+
+    private final Session session;
+    private final ByteArrayOutputStream output;
+    private final String nodePath;
+    private final String mimeType;
+
+    public FileNodeBackedOutputStream(SlingRepository repository, String nodePath, String mimeType) throws RepositoryException {
+        session = repository.loginService(null, null);
```

Comment:
this is not good, it should run with a session tied to the user that runs the export.